src/lib/factories/PathStartFactory.ts

```diff
@@ -0,0 +1,31 @@
+import path from 'node:path';
+import * as fg from 'fast-glob';
+import {getMessage, BundleName} from '../messages';
+
+export class PathStartFactory {
+	public async create(pathStarts?: string[]): Promise<string[]|undefined> {
+		if (pathStarts == null) {
+			return pathStarts;
+		}
+		const processedPaths: string[] = [];
+		for (const pathStart of pathStarts) {
+			// Globs assume the path separator character is '/' and treat '\\' as an escape character. So on OSs where
+			// the path separator character isn't '/', we need to convert paths into Glob syntax before checking whether
+			// they're actually Globs, to prevent misidentification.
+			const normalizedPathStart: string = path.sep === '/' ? pathStart : fg.convertPathToPattern(pathStart);
```

Comment:
This ternary is necessary because calling `convertPathToPattern` in POSIX on a well-formed glob actually mangles the glob.

Reply:
Can you explain this a bit more. I would have assumed that we could just call fg.convertPathToPattern(...) universally. It looks like it does things like escape weird characters in names that can also happen on unix... like trying to all folders that have something like "/some/odd(folderWithParentheses)/*.txt"

Reply:
See [this link](https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#convertpathtopatternpath).
When you do `convertPathToPattern('./**/*')` on POSIX, it interprets the `*`s as special characters that need to be escaped, so you get `./\\*\\*/\\*`, which is a busted glob. When you do it on Windows, the path isn't mangled.
My theory is that it's using "path" to refer to an actual _path_, e.g., a path to the current folder, and that you'd convert that into a pattern and then append more pattern stuff onto it.
I'm fine using `path.replace()` too if that's what we'd prefer.

Reply:
As discussed in person... the issue comes with things that are both valid globs and valid paths that are not obvious globs. For example someFile_[abc].txt can either be interpreted as '[abc'] being a part of the path name or '[abc]' be interpreted as a glob sequence (which makes any of the characters in the sequence in order to match someFile_a.txt for example). If the user intended to actually pick up their 'someFile_[abc].txt' file but it was interpreted as a glob, then we would end up not including their file at all.

For this reason, my suggestion is to have us only support positive glob cases. That is, we simply check if the path has a * or a ? in it. If so, then we treat the whole thing as a glob. Otherwise, we treat it as a file path. If the user wants to make something like 'someFile_[abc].txt' act as a glob, then they can simply do 'someFile_[abc].txt*' if they wanted.

I also propose we explicitly check for `!` and if it is the first character, then we throw an error message saying we do not support negative glob patterns. And then we should discuss with the team to see if instead we should support negative globs or not or if we should wait for a valid use case.

Reply:
If in the future we want to support negative glob patterns... then we should do so with a separate work item. If we did want it, then the algorithm would simply separate the paths into 4 buckets:
* positive non-glob paths
* positive globs (contain * or ?)
* negative non-glob paths (start with ! but do not contain * or ?)
* negative globs (start with ! and contains * or ?)

Then we would use the convertPathToPattern on all the non-globs and add them to the globs. And we would use convertPathToPattern on the negative non-glove paths (except for the first ! character and then put the ! character back on) and add them to the negative globs. Then we would send everything to fast-glob to get the resulting list for us.

(If you are really ambitious - we could just do this right now and add in more tests for the negative cases)